cpp/src/db/db_object.cu

```diff
@@ -120,84 +120,62 @@ bool db_pattern<idx_t>::isAllConstants() {
 template class db_pattern<int32_t>;
 template class db_pattern<int64_t>;
 
-template<typename idx_t>
-void db_column_index<idx_t>::deleteData() {
-  if (offsets != nullptr) {
-    ALLOC_FREE_TRY(offsets, nullptr);
-    offsets = nullptr;
-    offsets_size = 0;
-  }
-  if (indirection != nullptr) {
-    ALLOC_FREE_TRY(indirection, nullptr);
-    indirection = nullptr;
-    indirection_size = 0;
-  }
-}
-
 template<typename idx_t>
 db_column_index<idx_t>::db_column_index() {
-  offsets = nullptr;
   offsets_size = 0;
-  indirection = nullptr;
   indirection_size = 0;
 }
 
 template<typename idx_t>
-db_column_index<idx_t>::db_column_index(idx_t* _offsets,
+db_column_index<idx_t>::db_column_index(rmm::device_buffer&& _offsets,
                                         idx_t _offsets_size,
-                                        idx_t* _indirection,
+                                        rmm::device_buffer&& _indirection,
                                         idx_t _indirection_size) {
```

Comment:
I recommend using trailing _ for class member variables and avoid using either preceding/trailing _ for function input parameters. There is a good chance that this will go to our coding guidelines.

And using preceding _ in variable names is highly discouraged (see https://stackoverflow.com/questions/3136594/naming-convention-underscore-in-c-and-c-sharp-variables/20496955#20496955).